src/cpp/rtps/reader/StatefulReader.cpp

```diff
@@ -569,7 +594,8 @@ bool StatefulReader::processDataFragMsg(
             // If this is the first time we have received fragments for this change, add it to history
             if (change_created != nullptr)
```

Comment:
Why don't use `can_change_be_added_nts` when accepting first fragment of a CacheChange_t?

Reply:
The `incomingChange` received on the argument has the payload size of the fragment, so we cannot pass it to `can_change_be_added_nts`. And calling it on the line where you have put your comment is not necessary, since `change_received` is being called.

Reply:
Sorry, I wrote the comment where Github allows me. I cannot write it in code without changes.

This function receives the `sampleSize`. Cannot be used? Extend your new function to support it.

Reply:
Complete refactor of the new function to just accept the necessary input